### PR TITLE
update intro tutorial for pandas and CSV and batch_run and windows

### PR DESCRIPTION
**Pull request by tpike3**
Updated tutorial. 

The .ipynb is still large mainly due to the cells being cleared but following this practice in the future should prevent this from happening again.

### CODE REVIEW

**Comment by rht** [comment]
You need to rebase your branch on top of the upstream main branch. Somehow the .pre-commit-config.yaml change is included in this PR.

**Comment by tpike3** [comment]
Fixed, I think I squashed to many commits.

**Comment by EwoutH** [comment]
Thanks for figuring this out, documenting it and adding it to the tutorial! This was a big issue on Windows.

There is no way we can fix this in the class/method source code itself, by the way?

File: docs/tutorials/intro_tutorial.rst [hunk 1]
```diff
@@ -852,6 +853,19 @@ Or to plot the wealth of a given agent (in this example, agent 14):
...
 
+You can also use pandas to export the data to a CSV (comma separated value),
+which can be opened by any common spreadsheet application or opened by pandas.
+
+If you do not specify a file path, the file will be saved in the local directory.
+After you run the code below you will see two files appear (*model_data.csv* and *agent_data.csv*)
+
+.. code:: python
+
+    #save the model data (stored in the pandas gini object) to csv
```
Comment (rht) [inline docs/tutorials/intro_tutorial.rst:864]: Need a space after the "#", as per PEP 8.

Comment (tpike3) [reply docs/tutorials/intro_tutorial.rst]: fixed

**Comment by tpike3** [comment]
> Thanks for figuring this out, documenting it and adding it to the tutorial! This was a big issue on Windows.
> 
> There is no way we can fix this in the class/method source code itself, by the way?

@EwoutH This is a non-trivial problem see #1194

**Comment by rht** [comment]
@EwoutH 's question seems to be specifically on whether the `freeze_support()` can be auto-called when the `batch_run()` is called. Reading from internet, people always suggest putting it right after `if __name__ == "__main__":`. It could work inside `batch_run()` as well, but I can't test it since I don't have access to a Windows machine.

File: docs/tutorials/intro_tutorial.rst [hunk 2]
```diff
@@ -852,6 +853,19 @@ Or to plot the wealth of a given agent (in this example, agent 14):
...
 
+You can also use pandas to export the data to a CSV (comma separated value),
+which can be opened by any common spreadsheet application or opened by pandas.
+
+If you do not specify a file path, the file will be saved in the local directory.
+After you run the code below you will see two files appear (*model_data.csv* and *agent_data.csv*)
+
+.. code:: python
+
+    # save the model data (stored in the pandas gini object) to csv
```
Comment (rht) [inline docs/tutorials/intro_tutorial.rst:864]: Was about to say `CSV`, but we have gone through too many back and forth. Later I will fix it across the repo once for all.

**Pull request merged**